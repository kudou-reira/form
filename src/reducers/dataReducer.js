import data from '../lib/dishes.json';

const initialState = {
	data: data
};

export default (state = initialState) => {
	return state;
}
