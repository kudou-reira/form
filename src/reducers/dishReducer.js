import {
  DISH_COLLECTION_UPDATE,
  ADD_DISH,
  DELETE_DISH,
  RECORD_DISHES,
  DISH_COLLECTION_RESET,
  VERIFY_DISHES
} from '../actions/types';

const INITIAL_STATE = {
	dishCollection: [],
	verifyDishes: false,
	recordDishes: 0
};

export default (state = INITIAL_STATE, action) => {
  switch (action.type) {
  	case ADD_DISH:
  		return {...state, dishCollection: [...state.dishCollection, action.payload]};
  	case DELETE_DISH:
  		let index = state.dishCollection.findIndex((dish) => dish.id === action.payload); 
  		return {...state, dishCollection: [...state.dishCollection.slice(0, index), ...state.dishCollection.slice(index+1)]};
  	case RECORD_DISHES:
  		return {...state, recordDishes: action.payload}
  	case VERIFY_DISHES:
      return {...state, verifyDishes: action.payload}
  	case DISH_COLLECTION_RESET:
  		return {...state, dishCollection: []}
  	case DISH_COLLECTION_UPDATE:
  		var eventKey = action.payload.eventKey;
  		var type = typeof(eventKey);
  		var id = action.payload.id;
  		const existsAt = state.dishCollection.findIndex(dish => dish.id === id);
  		if(existsAt === -1) {
  			if(type === "string") {
  				var tempObject = {
  					id: id,
  					dish: eventKey,
  					servings: 1
  				}
  				return {...state, dishCollection: [...state.dishCollection, tempObject]};
  			}
  		} else {
  			console.log("this is existsAt", existsAt);
  			// exists in the array
  			// process update
  			// !! IMPORTANT !!
  			// check if the eventKey is a string and if it's equal to the current one or not
  			if(type === "string") {
  				console.log("this is type", type);
					return {
		        ...state,
		        dishCollection: state.dishCollection.map(dish => dish.id === id ?
	            // transform the one with a matching id
	            { ...dish, dish: eventKey } : 
	            // otherwise return original dish
	            dish
		        ) 
			    };
  			} else if(type === "number") {
  				console.log("this is type", type);
					return {
		        ...state,
		        dishCollection: state.dishCollection.map(dish => dish.id === id ?
	            // transform the one with a matching id
	            { ...dish, servings: eventKey } : 
	            // otherwise return original dish
	            dish
		        ) 
			    };
  			}
  		}
    default:
      return state;
  }
}
