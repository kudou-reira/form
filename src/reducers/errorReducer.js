import {
  ERROR,
  ERROR_SERVING
} from '../actions/types';

export const initialState = {
	error: true,
	errorServing: true
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case ERROR:
  		return {...state, error: action.payload}
  	case ERROR_SERVING:
  		return {...state, errorServing: action.payload}
    default:
      return state;
  }
}
