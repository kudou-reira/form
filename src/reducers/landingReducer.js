import {
  SELECT_MEALTIME,
  SELECT_PEOPLE,
  VERIFY_LANDING
} from '../actions/types';

// export const initialState = {
//   hi: 'yes'
// }

export const initialState = {
	mealtime: '----',
	numberOfPeople: "----",
  verifyLanding: false
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case SELECT_MEALTIME:
  		return {...state, mealtime: action.payload}
  	case SELECT_PEOPLE:
  		return {...state, numberOfPeople: action.payload}
    case VERIFY_LANDING:
      return {...state, verifyLanding: action.payload}
    default:
      return state;
  }
}
