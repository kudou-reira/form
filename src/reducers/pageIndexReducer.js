import {
  PAGE_INDEX
} from '../actions/types';

const initialState = {
	page: 1
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case PAGE_INDEX:
  		return {...state, page: action.payload}
    default:
      return state;
  }
}
