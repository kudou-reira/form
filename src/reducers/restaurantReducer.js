import {
  SELECT_RESTAURANT,
  VERIFY_RESTAURANT
} from '../actions/types';

export const initialState = {
	restaurant: '----',
  verifyRestaurant: false
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case SELECT_RESTAURANT:
  		return {...state, restaurant: action.payload}
    case VERIFY_RESTAURANT:
      return {...state, verifyRestaurant: action.payload}
    default:
      return state;
  }
}
