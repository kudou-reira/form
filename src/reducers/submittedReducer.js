import {
  VERIFY_SUBMITTED
} from '../actions/types';

const initialState = {
	verifySubmitted: false
};

export default (state = initialState, action) => {
  switch (action.type) {
  	case VERIFY_SUBMITTED:
      return {...state, verifySubmitted: action.payload}
    default:
      return state;
  }
}
