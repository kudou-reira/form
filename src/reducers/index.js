import { combineReducers } from 'redux';
import landingReducer from './landingReducer';
import restaurantReducer from './restaurantReducer';
import pageIndexReducer from './pageIndexReducer';
import dataReducer from './dataReducer';
import dishReducer from './dishReducer';
import errorReducer from './errorReducer';

export default combineReducers({
	landing: landingReducer,
	restaurant: restaurantReducer,
	pageIndex: pageIndexReducer,
	data: dataReducer,
	dish: dishReducer,
	error: errorReducer
});

