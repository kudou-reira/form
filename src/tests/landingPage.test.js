import LandingPage from '../components/landingPage';
import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { initialState as LandingReducer } from '../reducers/landingReducer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('<LandingPage />', () => {

	// function mapStateToProps(state) {
	// 	return {
	// 		landing: state.landing,
	// 		data: state.data,
	// 		error: state.error,
	// 		restaurant: state.restaurant
	// 	}
	// }

	// console.log("this is map", mapStateToProps(reducers));

	console.log("this is landing!!", LandingReducer);


  it('renders with default props', () => {
    const store = mockStore(reducers);
    const wrapper = shallow(
      <LandingPage store={store} />
    );
    console.log("this is initialState", reducers);
    // console.log("this is wrapper props in first test", wrapper.props());
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('changes meal time to breakfast if breakfast is selected', () => {
  	const store = mockStore(reducers);
    wrapper = mount( <Provider store={store}><LandingPage /></Provider> );
    // data is undefined, why?

    console.log("this is the store", store);

    // wrapper.find("#mealtime").simulate('change',{ target: { value: "breakfast" } });
    // console.log("this is wrapper instance", wrapper.props().selectMealtime('breakfast'));
    // wrapper.update();
    // console.log("this is wrapper instance", wrapper.props());
		// expect(wrapper.props().mealtime).toBe("breakfast");
  });
});


