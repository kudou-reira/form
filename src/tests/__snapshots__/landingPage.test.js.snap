// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage /> renders with default props 1`] = `
<LandingPage
  addDish={[Function]}
  deleteDish={[Function]}
  dishCollectionReset={[Function]}
  dishCollectionUpdate={[Function]}
  recordDishes={[Function]}
  selectMealtime={[Function]}
  selectPeople={[Function]}
  selectRestaurant={[Function]}
  sendError={[Function]}
  sendErrorServing={[Function]}
  sendPageIndex={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  verifyDishes={[Function]}
  verifyLanding={[Function]}
  verifyRestaurant={[Function]}
  verifySubmitted={[Function]}
/>
`;
